### DISCUSSION TITLE
Great app but need some features

### DISCUSSION THREAD

**Discussion by Frenchy614** [madmustachecompany/Blast-Workout-App#33] [Ideas]
### Summary

To start off I am a big fan of your app. I've been searching a long time and so far your app is the best. It is clear, simple, pretty and the information given is pretty useful. I am a premium user and I can't stop telling my friends at the gym about it.

It is in this spirit that I am writing to you. I think user feedback is very important for companies to grow. 

There are a few things that you need to improve. some of them are maybe possible but i did not find the way of doing so, you’ll have to forgive me. 

I am French, and the translation is pretty good, but some titles are wrong. We should be able to change the name of the exercise locally. 

When you create a training or a routine, be able to change the order of exercises and not just replace them. I sometimes find that some exercises are better at one time or change when I do my exercises in my routine all together. 

Add time for planking and not just reps and charge. 

When you do a routine, also add physical rest time between exercises with a timer. For some exercises my rest time is different and also time between reps.

Been able to create our own exercices with your system of muscle location ( really good by the way)

Allow us to add pics to our exercises. We do not have the QR codes on machines here and sometimes taking a pic of the machine helps. 

Add for the body measure the % IMG ( Grease mass in english in believe) that gives an important measurement when you trying to lose weight or else. 

Am pushing but maybe add a cold plunge and sauna.

Again pushing but maybe add a scale from 1 to 10 on how we felt the difficulty of the exercise or a training session.

Again your app is very good, and I read on your blog all the updates that you want to do. You are on the path of being one of the best apps out there. 

Thanks for your work and dedication. 

### Basic Example

.

**Comment by noelielc** [comment]
Hi @Frenchy614 and welcome!

Thanks for your comment, we are really happy that you like and share the app :)
I have broken down the feature requests into several queries based on your comments, so that they can be addressed separately:
- #31
- #32 
- #28
- #29 
- #30

Some of them require further discussion to understand their scope and feasibility more precisely:

(#29) _"add pics to our exercises"_
I'm not sure if I understood the request correctly. Can you please explain a little more?

(#30) _"add physical rest time between exercises with a timer"_
Can you give more details on what you are looking for?
Here is what I do personally if I want a longer or shorter rest time than the default timer. I press the desired time (for example 2:30 or 0:30) before validating my set in order to start the timer (which inhibits my default timer of 1:30). You can set up to 4 different timers, you can also quickly add or remove 15s or 30s once the timer is started.

_"french translation is pretty good, but some titles are wrong"_
The choice of French exercise names is tricky because English names are increasingly used (e.g. Squat or Curl for the biceps). This is why the exercise database works with aliases (several names are associated to each exercise). So when you search for an exercise using the search bar, the list of available aliases associated with each exercise is also consulted. For example, if you search for Squat, you will find " Flexion de jambes parallèles " because Squat is an alias for this exercise.
Could you provide us with a list of the incorrect titles you have spotted?

_"how to add a cold plunge and sauna"_
Can you explain how this should work? Should this be used as an exercise on the workout view?

Finally, concerning the last two points which are existing features, here "How to":
- change the exercise order?
Long press on the exercise icon then drag and drop the exercise where you want it to be, this also works with supersets.
<img width="120" alt="image" src="https://user-images.githubusercontent.com/63174/223003022-aca5d9b1-1fa2-4b29-a0f7-a88b1d4577e5.png">

- add difficulty scale to set?
You can use the RPE scale to assess the difficulty of a set, however note that sets marked with RPE will be excluded from 1RM estimates. To do so, long press on the set checkbox and the RPE selector will show with a range from 6 to 10.
6 to 7 = hard,
8 to 9  = very hard
10 = maximum effort
<img width="277" alt="image" src="https://user-images.githubusercontent.com/63174/223002818-9abdfb9d-fdf9-481e-beb7-9902f42ec971.png">
If compare to the RIR (rep in reserve) scale: 7 RPE = 3 RIR | 8 RPE = 2 RIR | 9 REP= 1 RIR | 10 RPE = 0 RIR

**Comment by Frenchy614** [comment]
#29 For exemple, I am starting a new exercice, I can take a photo of my posture or machine and add it somewhere to remember it. Like that when I start my new routine or exercice i just can look at it. 
#30 I mean, when I plan a routine to be able to add specific times between my exercices and set. I, for exemple, do 4 set of bench press. first set 45" rest, second 45" rest, third 30" rest, fourth 15 " rest. and also been able to do that between exercices. Like that when i finish my set or exercices i know automatically my rest time without having to play to much with the app. Of course leave the possibility to add time if needed like it is the case now. 

For the french translation don't worry about it, I speak english and i'll manage. But letting me change the name or add a note to the exercice locally would be cool. It's a plus since it is missing many exercices. 

For cold plunge or sauna just has an exercice with time and temperature. This type of exercices is getting more and more popular and the benefits are tremendous. On top of that your app is a tracking app the more I can track on it the happier I am.

For the last 2 thank you, I was not aware of it. 

Again, i really like your app, and the people at the gym are also enjoying it. Really happy to be a premium member, can't wait to see what's cooking next. Thank you for your dedication.
